lib/Sema/TypeCheckAvailability.h

```diff
@@ -68,6 +69,29 @@ enum class ExportabilityReason : unsigned {
   ExtensionWithConditionalConformances
 };
 
+/// A description of the restrictions on what declarations can be referenced
+/// from a the signature or body of a declaration.
+///
+/// We say a declaration is "exported" if it is `public` or
+/// `@usableFromInline`, not `_@spi`, and not visible via an
+/// `@_implementationOnly` import.
+///
+/// The "signature" of a declaration is the set of all types written in the
+/// declaration (such as function parameter and return types), but not
+/// including the function body.
+///
+/// The signature of an exported declaration can only reference other
+/// exported types.
+///
+/// The body of an inlinable function can only reference other `public` and
+/// `@usableFromInline` declarations; furthermore, if the inlinable
+/// function is also exported, its body is restricted to referencing other
+/// exported declarations.
```

Comment:
When you say *if the inlinable function is also exported*, that would be for all inlinable functions except `@_spi` right?

Reply:
Yep